x-pack/plugins/telemetry_collection_xpack/schema/xpack_plugins.json

```diff
@@ -8855,6 +8855,2464 @@
                       }
                     }
                   }
+                },
+                "detection_rule_status": {
+                  "properties": {
+                    "all_rules": {
+                      "properties": {
```

Comment:
Oh my gosh why are there so many duplications of this schema?? I hope this one is generated at least?

Reply:
It is generated with:

```sh
 node scripts/telemetry_check.js --fix
```

Reply:
Ahh, I see there's a README.md in the root folder (`telemetry_collection_xpack/schema/README.md`). That appears to be out of date though (says 2 files when there's 3?), and doesn't include that script you just mentioned to generate it...it just says

> The X-Pack related schema for the content that will be nested in `stack_stats.kibana.plugins`.    
  It is automatically generated by `@kbn/telemetry-tools` based on the `schema` property provided by all the registered Usage Collectors via the `usageCollection.makeUsageCollector` API.

It does link off to a few other README's from there, but I did see it mentioned in those. Though searching for that command shows it's in `src/plugins/usage_collection/README.mdx` & `packages/kbn-telemetry-tools/GUIDELINE.md`. Hoochie moochie this is hard to follow. Is there a better/recommended point of entry for groking all this?

Reply:
That is Kibana core's files.